builtin/credential/ldap/path_login.go

```diff
@@ -46,6 +47,13 @@ func pathLogin(b *backend) *framework.Path {
 
 func (b *backend) pathLoginAliasLookahead(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
 	username := d.Get("username").(string)
+	cfg, err := b.Config(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	if !*cfg.CaseSensitiveNames {
+		username = strings.ToLower(username)
+	}
```

Comment:
Can we add some tests to cover the cases above?

Reply:
added in https://github.com/hashicorp/vault/pull/31352/commits/3b2636f111d37f6185224e04ba71e71bc50db337